samples/client/petstore/java/jersey2/src/main/java/io/swagger/client/ApiClient.java

```diff
@@ -102,6 +102,15 @@ public JSON getJSON() {
     return json;
   }
 
+  public Client getHttpClient() {
```

Comment:
Jason, I originally wrote the issue #2849 . It would be useful in your PR to change the buildHttpClient() method to protected instead of private, allowing a subclass to override it.